Mathlib/FieldTheory/PrimitiveElement.lean

```diff
@@ -214,6 +256,157 @@ noncomputable def powerBasisOfFiniteOfSeparable : PowerBasis F E :=
 
 end SeparableAssumption
 
+section FiniteIntermediateField
+
+theorem exists_primitive_element_of_finite_intermediateField :
+    Finite (IntermediateField F E) → ∃ α : E, F⟮α⟯ = ⊤ := by
+  intro h
+  rcases isEmpty_or_nonempty (Fintype F) with (F_inf | ⟨⟨F_finite⟩⟩)
+  · let P : IntermediateField F E → Prop := fun K => ∃ α : E, F⟮α⟯ = K
+    have base : P ⊥ := ⟨0, adjoin_zero⟩
+    have ih : ∀ (K : IntermediateField F E) (x : E), P K → P (K⟮x⟯.restrictScalars F) := by
+      intro K β hK
+      cases' hK with α hK
+      rw [← hK, adjoin_simple_adjoin_simple]
+      haveI : Infinite F := isEmpty_fintype.mp F_inf
+      cases' primitive_element_inf_aux_of_finite_intermediateField F α β with γ hγ
+      exact ⟨γ, hγ.symm⟩
+    exact induction_on_adjoin P base ih ⊤
+  · exact exists_primitive_element_of_finite_bot F E
+
+lemma finite_intermediateField_of_exists_primitive_element.aux_finrank
+    (K : IntermediateField F E) :
+    letI : Algebra K (⊤ : IntermediateField K E) := IntermediateField.algebra _
+    letI : Module K (⊤ : IntermediateField K E) := Algebra.toModule
+    finrank K (⊤ : IntermediateField K E) = finrank K E := by
+  letI : Algebra K (⊤ : IntermediateField K E) := IntermediateField.algebra _
+  letI : Module K (⊤ : IntermediateField K E) := Algebra.toModule
+  apply LinearEquiv.finrank_eq
+  exact {
+    toFun := fun x => x.1,
+    map_add' := fun x y => by simp only [IntermediateField.coe_add],
+    map_smul' := fun r x => by
+      simp only [AddHom.toFun, RingHom.id_apply]
+      letI : SMul K K := Algebra.toSMul (self := Algebra.id K)
+      letI : IsScalarTower K K E := IsScalarTower.left K
+      exact IntermediateField.coe_smul (⊤ : IntermediateField K E) r x,
+    invFun := fun x => ⟨x, mem_top⟩,
+    left_inv := fun x => by simp only,
+    right_inv := fun x => by simp only,
+  }
+
+lemma finite_intermediateField_of_exists_primitive_element.aux_1
+    (S : Set E) (hprim : adjoin F S = ⊤) (K : IntermediateField F E) :
+    adjoin K S = ⊤ := by
+  apply restrictScalars_injective (K := F) (L' := K) (L := E)
+  rw [restrictScalars_top, ← top_le_iff, ← hprim, adjoin_le_iff,
+    coe_restrictScalars, ← adjoin_le_iff]
+
+lemma finite_intermediateField_of_exists_primitive_element.aux_2
```

Comment:
I don't think these are good names, if these lemmas would be useful for any other proof they should get descriptive names, if not they should be `private`

Reply:
I don't think they are useful in other places *yet*. The `aux_1` may be more useful, but let me make both of them `private` for now.